IdentityCore/tests/MSIDKeyOperationUtilTest.m

```diff
@@ -0,0 +1,251 @@
+    NSData *rsaDataSignature = [dataDigest msidSignHashWithPrivateKey:self.rsaPrivateKey];
+    XCTAssertEqualObjects([rsaDataSignature msidBase64UrlEncodedString], [signature msidBase64UrlEncodedString]);
+
+}
+
+#pragma mark -- Test Utility
+
+- (void)populateTestKeys
+{
+    [self fetchKeys];
+    if (self.eccPublicKey == NULL)
+    {
+        [self generateKeyPair];
+    }
+}
+
+- (void)fetchKeys
+{
+    NSData *tag = [self.sharedAccessGroup dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableDictionary * queryPrivateKey = [[NSMutableDictionary alloc] init];
+
+    // Set the private key query dictionary.
+    [queryPrivateKey setObject:(__bridge id)kSecClassKey forKey:(__bridge id)kSecClass];
+    [queryPrivateKey setObject:tag forKey:(__bridge id)kSecAttrApplicationTag];
+    [queryPrivateKey setObject:(__bridge id)kSecAttrKeyTypeEC forKey:(__bridge id)kSecAttrKeyType];
+    [queryPrivateKey setObject:[NSNumber numberWithBool:YES] forKey:(__bridge id)kSecReturnRef];
+    OSStatus status= noErr;
+    SecKeyRef privateKeyReference = NULL;
+    status = SecItemCopyMatching((__bridge CFDictionaryRef)queryPrivateKey, (CFTypeRef *)&privateKeyReference);
+    
+    if (status != errSecSuccess)
+    {
+        privateKeyReference = nil;
+    }
+
+    self.eccPrivateKey = privateKeyReference;
+    if (self.eccPrivateKey != NULL)
+    self.eccPublicKey = SecKeyCopyPublicKey(self.eccPrivateKey);
+}
+
+- (void)generateKeyPair
+{
+    SecAccessControlRef access = NULL;
+    if (@available(macOS 10.12.1, iOS 9.0, *))
+    {
+        access = SecAccessControlCreateWithFlags(kCFAllocatorDefault,
+                                                 kSecAttrAccessibleWhenUnlockedThisDeviceOnly,
+                                                 kSecAccessControlPrivateKeyUsage,
+                                                 NULL);
+    }
+    
+    NSData *tag = [self.sharedAccessGroup dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary* attributes =
+      @{ (id)kSecAttrKeyType:(id)kSecAttrKeyTypeECSECPrimeRandom,
+         (id)kSecAttrKeySizeInBits:@256,
+         (id)kSecAttrTokenID:(id)kSecAttrTokenIDSecureEnclave,
+         (id)kSecPrivateKeyAttrs:
+             @{
+                 (id)kSecAttrIsPermanent:@YES,
+                 (id)kSecAttrApplicationTag:tag,
+                 (id)kSecAttrAccessControl:(__bridge id)access,
+                 (id)kSecAttrCanEncrypt:@NO
+             }
+       };
+    
+    CFErrorRef error = NULL;
+    SecKeyRef privateKey = SecKeyCreateRandomKey((__bridge CFDictionaryRef)attributes,
+                                                 &error);
+    if (!privateKey) {
+        NSError *err = CFBridgingRelease(error);
+        XCTAssertNotNil(err);
+        err = nil;
+        return;
+    }
+    XCTAssertTrue(error == NULL);
+    self.eccPrivateKey = privateKey;
+    self.eccPublicKey = SecKeyCopyPublicKey(self.eccPrivateKey);
```

Comment:
nit SecKeyCopyPublicKey retains it, so you might want to release it later.

Reply:
Good catch, will release it in teardown